### ISSUE TITLE
VS 2017 freezes on delete commands

### ISSUE THREAD

**Issue by dpippert** [VsVim/VsVim#2777]
Most of the time, doing vim delete commands such as: dd. D
causes Visual Studio to completely freeze and it has to be restarted.
Using VS 2017 Version 15.7.3 with .NET Framework 4.8.03752

I have all keys mapped to Vim mode if that matters in Tools > Options > VsVim

1. Open a file.
2. Navigate to any line in the file.
3. Type "dd" in Vim command mode.
4. Hope for the best, but frequently VS 2017 freezes and has to be restarted.

I cannot figure out why sometimes it freezes and sometims not.

Files are tiny, can be 10 lines and it still freezes.

- Visual Studio version: 2017 V. 15.7.3 with .NET Framework 4.8.03752
- VsVim version: 2.8.0.0
- Programming Language: seems to be any

- Check(Type 'x') any that are installed:
no other tools installed that I know of

![vs_freeze_dd](https://user-images.githubusercontent.com/7669192/73982737-d4f7e700-4902-11ea-9bd8-bf379a53fda9.png)
General Keyboard configuration problems are described below.
https://github.com/VsVim/VsVim/issues/2527

**Comment by ricksladkey** [comment]
Do you have `:set clipboard=unnamed` in effect? If so, can you check whether yank exhibits the same symptoms? If it does, then the problem might be the clipboard.

**Comment by dpippert** [comment]
How do I see options. I tried :se all
but it says not supported. I then did :se clipboard?
and it says clipboard=""

yy is something I do very often so I would have seen that freeze before.
Seems to be dd and right now getting superstitious, would it have
anything to do with the file type? The file the freeze has occurred on last
couple days is .json, in fact here it is below...

{
  "$schema": "
https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#
",
  "contentVersion": "1.0.0.0",
  "parameters": {
    "env": {
      "defaultValue": "dev",
      "type": "string"
    }
  },
  "variables": {
    "route-table":"[concat('ead-', parameters('env'), '-sha-route-table')]"
  },
  "resources": [
    {
      "type": "Microsoft.Network/routeTables",
      "apiVersion": "2019-11-01",
      "name": "[concat('ead-', parameters('env'), '-sha-route-table')]",
      "location": "eastus2",
      "tags": {
        "displayName": "UDR - Subnet"
      },
      "properties": {
        "disableBgpRoutePropagation": false,
        "routes": [
          {
            "name": "route-ASE-0",
            "properties": {
              "addressPrefix": "0.0.0.0/0",
              "nextHopType": "Internet"
            }
          }
        ]
      }
    },
    {
      "type": "Microsoft.Network/routeTables/routes",
      "apiVersion": "2019-11-01",
      "name": "[concat(variables('route-table'), '/route-ASE-0')]",
      "dependsOn": [
        "[resourceId('Microsoft.Network/routeTables',
variables('route-table'))]"
      ],
      "properties": {
        "addressPrefix": "0.0.0.0/0",
        "nextHopType": "Internet"
      }
    }
  ]
}

Appreciate the reply, new ideas shoot my way I'll try it out.

Also I often would do something like select a block of text in view mode,
then "*p to put the selected text on the clipboard. Have not seen that act
up either, always works. I do recall at least once, that dd against a view
mode selected block also froze VS 2017.

**Comment by ricksladkey** [comment]
You can type `:set clipboard?` to see the value of the `clipboard` setting

**Comment by dpippert** [comment]
Yes my earlier email said I did :se clipboard? and it came back ""

This is weird. It seems like "dd" against the first line in the file is
certain death to Visual Studio 2017. I thought it was not that specific,
but if you are trying to reproduce this, do dd against the FIRST line of
the .json file.
